compiler/modules/nback/number.bal

```diff
@@ -249,6 +277,49 @@ function buildConvertDecimalToFloat(llvm:Builder builder, Scaffold scaffold, llv
     buildStoreFloat(builder, scaffold, result, insn.result);
 }
 
+function buildConvertToDecimal(llvm:Builder builder, Scaffold scaffold, bir:ConvertToDecimalInsn insn) returns BuildError? {
+    var [repr, val] = check buildReprValue(builder, scaffold, insn.operand);
+    if repr.base == BASE_REPR_INT {
+        buildConvertIntToDecimal(builder, scaffold, val, insn);
+        return;
+    }
+    else if repr.base == BASE_REPR_FLOAT {
+        buildConvertFloatToDecimal(builder, scaffold, val, insn);
+        return;
+    }
+    // convert to decimal from tagged pointer
+    llvm:Value resultWithErr = <llvm:Value>builder.call(scaffold.getRuntimeFunctionDecl(convertToDecimalFunction), [val]);
+    buildStoreDecimal(builder, scaffold, buildCheckPanicCode(builder, scaffold, resultWithErr, insn), insn.result);   
+    return;
+}
+
+function buildConvertIntToDecimal(llvm:Builder builder, Scaffold scaffold, llvm:Value intVal, bir:ConvertToDecimalInsn insn) {
+    llvm:BasicBlock continueBlock = scaffold.addBasicBlock();
+    llvm:BasicBlock errBlock = scaffold.addBasicBlock();
+    builder.condBr(llvm:constInt(LLVM_BOOLEAN, 0), errBlock, continueBlock);
+    builder.positionAtEnd(errBlock);
+    builder.br(scaffold.getOnPanic());
+    builder.positionAtEnd(continueBlock);
+    buildStoreDecimal(builder, scaffold, <llvm:Value>builder.call(scaffold.getRuntimeFunctionDecl(decimalFromIntFunction), [intVal]), insn.result);
+}
+
+function buildConvertFloatToDecimal(llvm:Builder builder, Scaffold scaffold, llvm:Value floatVal, bir:ConvertToDecimalInsn insn) {
+    llvm:Value resultWithErr = <llvm:Value>builder.call(scaffold.getRuntimeFunctionDecl(decimalFromFloatFunction), [floatVal]);
+    buildStoreDecimal(builder, scaffold, buildCheckPanicCode(builder, scaffold, resultWithErr, insn), insn.result);   
+}
+
+function buildCheckPanicCode(llvm:Builder builder, Scaffold scaffold, llvm:Value valWithErr, bir:ConvertToDecimalInsn insn) returns llvm:Value {
+    llvm:BasicBlock continueBlock = scaffold.addBasicBlock();
+    llvm:BasicBlock errBlock = scaffold.addBasicBlock();
+    llvm:Value panicCode = builder.extractValue(valWithErr, 1);
+    builder.condBr(builder.iCmp("ugt", panicCode, llvm:constInt("i64", 0)), errBlock, continueBlock);
```

Comment:
Elsewhere do we use ugt to check panic codes?  I would expect to just check for !=.

Reply:
Yes, change those places too